gsva/table_visualization.R

```diff
@@ -104,7 +93,15 @@ make.box <- function(data, sample, feature.name=NA, sample.name=NA, value.name=N
 }
 
 # Produce a clustered heatmap for the data
-make.heatmap <- function(data, colors=colorRampPalette(c("#2166ac", "#FFFFFF", "#b2182b")), breaks=seq(-1,1,by=0.02)) {
-    size = 4
-    hm <- pheatmap(data, color=colors(length(breaks)),border_color=NA, cluster_cols=TRUE, breaks=breaks, cellwidth=size, cellheight=size, fontsize=size)
+make.heatmap <- function(data, path, colors=colorRampPalette(c("#2166ac", "#FFFFFF", "#b2182b")), breaks=seq(-1,1,by=0.02), size=4) {
+    hm <- pheatmap(data, 
+                   filename=path,
```

Comment:
Very minor concern. When pheatmap saves the file, the legend shows breaks instead of being continuous color. Could consider saving with `png()` instead. Something like:
```
  png(path, width = 1200, height = 1200, res = 240)
  grid::grid.newpage()
  grid::grid.draw(hm$gtable)
  dev.off()
```

Reply:
Incorporated. The legend is still segmented into discrete colors, however the gaps between each segment have been eliminated and the image dimensions set based on the number of rows/columns. Will be visible after the next push.